src/environment_setup/netmhc.ml

```diff
@@ -0,0 +1,195 @@
+open Biokepi_run_environment
+open Common
+
+(* 
+  Tested against:
+
+    netMHC-4.0a.Linux.tar.gz
+    pickpocket-1.1a.Linux.tar.gz
+    netMHCpan-3.0a.Linux.tar.gz
+    netMHCcons-1.1a.Linux.tar.gz
+
+  Do not use custom named archives
+  and keep them as they are after you
+  download (i.e. no need to insert the
+  data folder in them or customize the
+  binaries)
+*)
+type netmhc_file_locations = {
+  netmhc: Workflow_utilities.Download.tool_file_location;
+  netmhcpan: Workflow_utilities.Download.tool_file_location;
+  pickpocket: Workflow_utilities.Download.tool_file_location;
+  netmhccons: Workflow_utilities.Download.tool_file_location;
+}
+
+(*
+  The standard netMHC installation requires
+  customizing some of the environment variables
+  defined in their main binary files. The following
+  function handles these replacements.
+*)
+let replace_env_value file envname newvalue =
+  let escape_slash txt =
+    let escfun c = if c = '/' then ['\\'; c] else [c] in
+    String.rev txt
+    |> String.fold ~init:[] ~f:(fun x c -> (escfun c) @ x)
+    |> List.map ~f:String.of_character
+    |> String.concat
+  in
+  let file_org = file in
+  let file_bak = file_org ^ ".bak" in
+  KEDSL.Program.(
+    shf "mv %s %s" file_org file_bak &&
+    shf "sed -e 's/setenv\t%s\t.*/setenv\t%s\t%s/g' %s > %s"
+      envname envname (escape_slash newvalue) file_bak file_org &&
+    shf "rm -f %s" file_bak
+  )
+
+(* 
+  e.g. input: /path/to/netMHC-3.4a.Linux.tar.gz
+  e.g. output: netMHC-3.4
+
+  This guessing game is necessary to build the URL
+  for the data files required for a complete installation
+  and also for to know what folder gets extracted from the
+  archives.
+*)
+let guess_folder_name tool_file_loc =
+  let loc = match tool_file_loc with
+    | `Scp l -> l
+    | `Wget l -> l
+    | `Fail _ -> "NoFile-0.0b.Linux.tar.gz"
+  in
+  let chop_final_char s =
+    let ssub = String.sub s 0 ((String.length s) - 1) in
+    match ssub with
+    | Some txt -> txt
+    | None -> s
+  in
+  loc (* /path/to/netMHC-3.4a.Linux.tar.gz *)
+    |> Filename.basename (* netMHC-3.4a.Linux.tar.gz *)
+    |> Filename.chop_extension (* netMHC-3.4a.Linux.tar *)
+    |> Filename.chop_extension (* netMHC-3.4a.Linux *)
+    |> Filename.chop_extension (* netMHC-3.4a *)
+    |> chop_final_char (* netMHC-3.4 *)
```

Comment:
Could you wrap all these in a 

```ocaml
try
   (* ... *)
with _ ->
   ksprintf failwith "Error while guessing folder name from %S" loc
```

so that we have better error messages :)